### PR TITLE
fix: PREV-51 - Fix 500 when input file empty

### PR DESCRIPTION
**Pull request by RakuJa**
Fixed PREV51, clean up log, update tests

### CODE REVIEW

File: app/core/routers/document.py [hunk 1]
```diff
@@ -58,7 +62,13 @@ async def get_preview(
     :return: 400 if there were invalid parameters, otherwise
     the requested file converted accordingly to pdf.
     """
-    if first_page >= 1 and (first_page <= last_page or last_page == 0):
+    validation_errors: Optional[Response] = check_for_document_metadata_errors(
+        first_page=first_page,
+        last_page=last_page,
```
Comment (federicorispo) [inline app/core/routers/document.py:67]: The default values for first_page and last_page are `1` and `0` and when the system checks this values it return `NUMBER_OF_PAGES_NOT_VALID`. Is this the intended behaviour?

File: app/core/routers/document.py [hunk 2]
```diff
@@ -98,7 +103,13 @@ async def post_preview(
     the requested file converted accordingly to pdf.
     """
 
-    if first_page >= 1 and (first_page <= last_page or last_page == 0):
+    validation_errors: Optional[Response] = check_for_document_metadata_errors(
+        first_page=first_page,
+        last_page=last_page,
```
Comment (federicorispo) [inline app/core/routers/document.py:108]: Same doubt as above :thinking:

File: app/core/routers/pdf.py [hunk 3]
```diff
@@ -58,7 +62,13 @@ async def get_preview(
     :return: 400 if there were invalid parameters, otherwise
     the requested pdf divided accordingly.
     """
-    if first_page >= 1 and (first_page <= last_page or last_page == 0):
+    validation_errors: Optional[Response] = check_for_document_metadata_errors(
```
Comment (federicorispo) [inline app/core/routers/pdf.py:65]: Same doubt as above :thinking:

File: app/core/routers/pdf.py [hunk 4]
```diff
@@ -98,7 +103,14 @@ async def post_preview(
     the requested pdf divided accordingly.
     """
 
-    if first_page >= 1 and (first_page <= last_page or last_page == 0):
+    validation_errors: Optional[Response] = check_for_document_metadata_errors(
+        first_page=first_page,
+        last_page=last_page,
+    )
```
Comment (federicorispo) [inline app/core/routers/pdf.py:109]: Same doubt as above :thinking:

**Review by federicorispo — CHANGES_REQUESTED** [review]

File: app/core/routers/document.py (hunk 1 above)
Comment (RakuJa) [reply app/core/routers/document.py:67]: it returns None if there are no errors (aka 1 first page and 0 last page)

Comment (federicorispo) [reply app/core/routers/document.py:67]: @RakuJa explains the logic of the `check_for_document_metadata_errors` function and the default values are working as intended!

**Review by federicorispo — APPROVED** [review]

**Pull request merged**